pysages/colvars/core.py

```diff
@@ -9,15 +9,16 @@
 
 from abc import ABC, abstractmethod
 from inspect import signature
-from typing import Callable, List, Tuple, Union
+from typing import Callable, Sequence, Tuple, Union
 
 from jax import grad as jax_grad, jit, numpy as np
+import numpy
 
 from pysages.utils import JaxArray, dispatch
 
 
 UInt32 = np.uint32
-Indices = Union[int, range]
+Indices = Union[numpy.intp, numpy.int, int, range]
```

Comment:
Isn't `numpy.intp` enough here?

Reply:
yes, numpy.intp is enough.
But I wasn't sure what else ppl might throw at PySAGES.
Could be a custom created array with the indices.

Does it hurt here?

Reply:
I'm asking because `numpy.int` got deprecated in numpy `1.20` and `numpy.intp` should point to the default precision. I think what you might want is `numpy.intc` and `numpy,int_` instead.

Reply:
oh, ok I wasn't aware of that.
Let's add all of them instead of the deprecated one?